go/enclave/enclave.go

```diff
@@ -448,6 +444,15 @@ func (e *enclaveImpl) ingestL1Block(br *common.BlockAndReceipts, isLatest bool)
 		// Unsure what to do here; block has been stored
 	}
 
+	// the sequencer has the task of duplicating the batches that have become non-canonical
+	sequencer, ok := e.service.(nodetype.Sequencer)
```

Comment:
I think we should avoid checking the service type and use another pattern, the enclave code was quite polluted because of this previously. I'm happy to have validators have DuplicateBatches too and do nothing just to avoid this check.